packages/runtime-core/__tests__/apiCreateComponent.spec.tsx

```diff
@@ -88,6 +88,22 @@ test('type inference w/ optional props declaration', () => {
   ;<Comp msg="hello" />
 })
 
+test('type inference for props in setup', () => {
+  const Comp = createComponent(
+    (_props: { a: string | number; b: object; c: number }) => () => h('div')
+  )
+  ;<Comp a="a" b={{}} c={1} />
+  ;<Comp a={1} b={{}} c={1} />
+})
+
+test('optional props in setup', () => {
+  const Comp = createComponent(
+    (_props: { a?: string; b?: object; c: number }) => () => h('div')
+  )
+  ;<Comp c={1} />
+  ;<Comp a="a" c={1} />
```

Comment:
why not `expect(<Comp a="a" b={{}} c={1} />).not.toThrow()`

Reply:
`toThrow` catches JS runtime exceptions, __not__ TS compile-time errors, so type errors will fail test compilation